presto-main/src/test/java/io/prestosql/operator/BenchmarkUnnestOperator.java

```diff
@@ -0,0 +1,424 @@
+    {
+        DriverContext driverContext = context.createTaskContext().addPipelineContext(0, true, true, false).addDriverContext();
+        Operator operator = context.getOperatorFactory().createOperator(driverContext);
+
+        Iterator<Page> input = context.getPages().iterator();
+        ImmutableList.Builder<Page> outputPages = ImmutableList.builder();
+
+        boolean finishing = false;
+        for (int loops = 0; !operator.isFinished() && loops < 1_000_000; loops++) {
+            if (operator.needsInput()) {
+                if (input.hasNext()) {
+                    Page inputPage = input.next();
+                    operator.addInput(inputPage);
+                }
+                else if (!finishing) {
+                    operator.finish();
+                    finishing = true;
+                }
+            }
+
+            Page outputPage = operator.getOutput();
+            if (outputPage != null) {
+                outputPages.add(outputPage);
+            }
+        }
+
+        return outputPages.build();
+    }
+
+    /**
+     * Helper class for random input generation.
+     */
+    public static class InputGenerator
+    {
+        private static final int INT_UPPER_LIMIT = 1_000_000_000;
+
+        private Random random = new Random(456);
+
+        // lower and upper bounds on nested structures in the input
+        private final int minNestedCardinality;
+        private final int maxNestedCardinality;
+
+        // upper bound on length of strings generated for VARCHAR elements
+        private final int maxStringLength;
+
+        public InputGenerator(int minNestedCardinality, int maxNestedCardinality, int maxStringLength)
+        {
+            checkArgument(minNestedCardinality >= 0 && maxNestedCardinality >= 0 && minNestedCardinality <= maxNestedCardinality, "cardinality bounds are invalid");
+            checkArgument(maxStringLength >= 0, "invalid string length");
+            this.minNestedCardinality = minNestedCardinality;
+            this.maxNestedCardinality = maxNestedCardinality;
+            this.maxStringLength = maxStringLength;
+        }
+
+        /**
+         * Generates a block with {@code entries} positions given an input {@code type}.
+         *
+         * {@code primitiveElementNulls} indicates what percentage of primitive elements
+         * (VARCHAR or INT) should be null. Similarly, the value {@code rowElementNulls}
+         * indicates the percentage for Row elements.
+         *
+         * The null percentages are not propagated inside a MapBlock's keyBlock. Everything
+         * is always non-null inside the returned keyBlock.
+         *
+         * The set S of valid input for {@code type} can be defined as:
+         * S = {VARCHAR, INTEGER, ROW(x), ARRAY(x), MAP(x, x)} where x belongs to S
+         *
+         */
+        public Block produceBlock(Type type, int entries, double primitiveElementNulls, double rowElementNulls)
```

Comment:
May be rename `primitiveElementNulls` and `rowElementNulls` to `xxxRatio` to make it more obvious what they represent (without having to look at the docs)

Reply:
Agreed, changed the variable names.